bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/internal/SaltGraphStyler.java

```diff
@@ -0,0 +1,204 @@
+ * limitations under the License.
+ * #L%
+ */
+
+package org.corpus_tools.hexatomic.graph.internal;
+
+import com.google.common.base.Joiner;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+import org.corpus_tools.salt.common.SDominanceRelation;
+import org.corpus_tools.salt.common.SPointingRelation;
+import org.corpus_tools.salt.common.SStructuredNode;
+import org.corpus_tools.salt.common.SToken;
+import org.corpus_tools.salt.core.SRelation;
+import org.corpus_tools.salt.graph.Label;
+import org.corpus_tools.salt.graph.LabelableElement;
+import org.corpus_tools.salt.util.SaltUtil;
+import org.eclipse.draw2d.Bendpoint;
+import org.eclipse.draw2d.ColorConstants;
+import org.eclipse.draw2d.Connection;
+import org.eclipse.draw2d.IFigure;
+import org.eclipse.draw2d.MidpointLocator;
+import org.eclipse.draw2d.ShortestPathConnectionRouter;
+import org.eclipse.draw2d.geometry.Dimension;
+import org.eclipse.draw2d.geometry.Point;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.zest.core.viewers.EntityConnectionData;
+import org.eclipse.zest.core.viewers.ISelfStyleProvider;
+import org.eclipse.zest.core.widgets.GraphConnection;
+import org.eclipse.zest.core.widgets.GraphNode;
+
+/**
+ * Defines colors and other styles for Salt graph elements.
+ * @author Thomas Krause
+ *
+ */
+public class SaltGraphStyler extends LabelProvider implements ISelfStyleProvider {
+
+  private final ShortestPathConnectionRouter pointingConnectionRouter;
+  private final IFigure figure;
+
+  public SaltGraphStyler(IFigure figure) {
+    this.figure = figure;
+    this.pointingConnectionRouter = new ShortestPathConnectionRouter(figure);
+  }
+
+  @Override
+  public String getText(Object element) {
+
+    if (element instanceof LabelableElement) {
+      LabelableElement node = (LabelableElement) element;
+      TreeMap<String, String> labelsByQName = new TreeMap<>();
+      for (Label l : node.getLabels()) {
+        boolean include = true;
+        if (element instanceof SRelation<?, ?> && "salt".equals(l.getNamespace())) {
+          include = false;
+        }
+        if ("salt".equals(l.getNamespace()) && "id".equals(l.getName())) {
+          include = false;
+        }
+        if (include) {
+          String qname = SaltUtil.createQName(l.getNamespace(), l.getName());
+          labelsByQName.put(qname, qname + "=" + l.getValue());
+        }
+      }
+      List<String> labels = new LinkedList<>(labelsByQName.values());
+
+      if (element instanceof SToken) {
+        SToken token = (SToken) element;
+        String coveredText = token.getGraph().getText(token);
+        labels.add(0, coveredText);
+      }
+
+      return Joiner.on('\n').join(labels);
+    }
+    if (element instanceof EntityConnectionData) {
+      return "";
+    }
+    throw new IllegalArgumentException("Object of type LabelableElement expectected, but got "
```

Comment:
```suggestion
    throw new IllegalArgumentException("Object of type LabelableElement expected, but got "
```

Another type correctectectected.